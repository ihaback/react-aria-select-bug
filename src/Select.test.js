import TestRenderer from "react-test-renderer";
import { AriaSelect } from "./Select";
import ReactDOM from "react-dom";

describe("Select", () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  test("generating snapshot for Select", () => {
    const testRenderer = TestRenderer.create(<AriaSelect />);

    expect(testRenderer).toMatchSnapshot();
  });
});
