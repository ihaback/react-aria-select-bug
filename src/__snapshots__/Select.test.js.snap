// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent generating snapshot for Select 1`] = `
Array [
  <div
    className="react-aria-Select"
    data-rac=""
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <span
      className="react-aria-Label"
      onClick={[Function]}
    >
      Favorite Animal
    </span>
    <button
      aria-controls={null}
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="react-aria-:r7:"
      className="react-aria-Button"
      data-rac=""
      id="react-aria-:r2:"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        className="react-aria-SelectValue"
        data-placeholder={true}
        data-rac=""
        id="react-aria-:r7:"
      >
        Select an item
      </span>
      <span
        aria-hidden="true"
      >
        â–¼
      </span>
    </button>
  </div>,
  <div
    aria-hidden={true}
    data-a11y-ignore="aria-hidden-focus"
    data-testid="hidden-select-container"
    style={
      Object {
        "border": 0,
        "clip": "rect(0 0 0 0)",
        "clipPath": "inset(50%)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
  >
    <input
      onFocus={[Function]}
      style={
        Object {
          "fontSize": 16,
        }
      }
      tabIndex={-1}
      type="text"
    />
    <label>
      <select
        onChange={[Function]}
        tabIndex={-1}
        value=""
      >
        <option />
      </select>
    </label>
  </div>,
]
`;
